web/app/components/dashboard/new-features-banner.ts

```diff
@@ -18,22 +19,39 @@ export default class DashboardNewFeaturesBanner extends Component<DashboardNewFe
    * until the user dismisses the banner.
    */
   protected get isShown(): boolean {
-    const storageItem = window.localStorage.getItem(LOCAL_STORAGE_ITEM_NAME);
+    /**
+     * If the banner has been dismissed, don't show it.
+     * This check causes the property to recompute when dismissed.
+     */
+    if (this.isDismissed) {
+      return false;
+    }
+
+    const storageItem = window.localStorage.getItem(
+      NEW_FEATURES_BANNER_LOCAL_STORAGE_ITEM
+    );
 
     if (storageItem === null) {
-      window.localStorage.setItem(LOCAL_STORAGE_ITEM_NAME, "true");
+      window.localStorage.setItem(
+        NEW_FEATURES_BANNER_LOCAL_STORAGE_ITEM,
+        "true"
+      );
       return true;
-    } else if (storageItem === "true" && !this.isDismissed) {
```

Comment:
Despite bindings to a tracked property, the `isShown` getter wasn't recomputing on `dismiss`.